### PR TITLE
fix(developer): kmcomp package compiler race condition

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes a race condition when compiling multiple packages, where the temp path may be re-used before the compiler has finished with it.

To take advantage of this fix, I wanted to make sure I was using the right kmcomp.exe when building unit tests for Keyman Core. This uncovered a couple of other things:

* Updates Keyman Core build infrastructure to use the current kmcomp.exe rather than a system-installed one.
* Corrects a bug that was identified by the 15.0 version of kmcomp in one unit test (duplicate `&NAME` stores -- the meson test script prepends a `&NAME` store to each source file during build).

@keymanapp-test-bot skip

### CODE REVIEW

File: common/core/desktop/tests/unit/kmx/028 - smp.kmn [hunk 1]
```diff
@@ -4,7 +4,6 @@ c expected: \U0001F601yx
 c context:
 
 store(&version) '10.0'
-store(&name) '028smp'
```
Comment (jahorton) [inline common/core/desktop/tests/unit/kmx/028 - smp.kmn:7]: Why this change?  It seems "thrown in."

Comment (mcdurdin) [reply common/core/desktop/tests/unit/kmx/028 - smp.kmn:7]: See original details above:
> * Corrects a bug that was identified by the 15.0 version of kmcomp in one unit test (duplicate &NAME stores -- the meson test script prepends a &NAME store to each source file during build).

Comment (mcdurdin) [reply common/core/desktop/tests/unit/kmx/028 - smp.kmn:7]: We've all apparently been building these keyboards with the 14.0 compiler until now.

File: common/core/desktop/tests/unit/kmx/meson.build [hunk 2]
```diff
@@ -77,7 +77,7 @@ tests = [
 ]
 
 if build_machine.system() == 'windows'
-  kmcomp = find_program('kmcomp.exe', required: false)
+  kmcomp = find_program(join_paths(meson.source_root(),'..','..','..','windows','bin','developer','kmcomp.exe'), 'kmcomp.exe', required: false)
```
Comment (jahorton) [inline common/core/desktop/tests/unit/kmx/meson.build:80]: Given your answer to my earlier comment...

> I wanted to make sure I was using the right kmcomp.exe when building unit tests for Keyman Core.

I didn't quite get that at the time, as the way it was worded made me think that there wasn't a related change here.  Looking at it again, though, I think the quote above relates to this line?

Comment (mcdurdin) [reply common/core/desktop/tests/unit/kmx/meson.build:80]: yep

**Review by jahorton — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.257-beta](https://keyman.com/downloads/releases/beta/15.0.257)
